test/new-e2e/tests/agent-subcommands/diagnose/diagnose_common_test.go

```diff
@@ -55,11 +102,23 @@ func getDiagnoseOutput(v *baseDiagnoseSuite, commandArgs ...agentclient.AgentArg
 func (v *baseDiagnoseSuite) TestDiagnoseDefaultConfig() {
 	diagnose := getDiagnoseOutput(v)
 	v.AssertOutputNotError(diagnose)
+
+	diagnose = getDiagnoseOutput(v, agentclient.WithArgs([]string{"--json"}))
+	diagnoseJSON := stringToJSON(diagnose)
```

Comment:
I don't the function name `stringToJSON` signal what the function is doing.

I would change the name to something that reflects what the function does. `unmarshalDiagnose` or something similar

Reply:
will do it thx for the idea